File: mirea/cources/2026-2027/aii-july/04/Part-01-Prefix-sums/examples/example-05.py
# Пример: Python-аналог exclusive_scan.
# Ввод: n и массив чисел.
# Вывод: массив префиксных сумм длины n + 1.

n = int(input())
a = list(map(int, input().split()))

extended = a + [0]
pref = [0] * (n + 1)
current = 0
for i in range(n + 1):
    pref[i] = current
    current += extended[i]

print(*pref)

File: mirea/cources/2026-2027/aii-july/04/Part-01-Prefix-sums/examples/example-05.cpp
// Пример: построение префиксных сумм через std::exclusive_scan.
// Ввод: n и массив int.
// Вывод: массив префиксных сумм длины n + 1.

#include <iostream>
#include <numeric>
#include <vector>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;

    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    // Добавляем фиктивный ноль, чтобы получить pref длины n + 1.
    vector<int> extended = a;
    extended.push_back(0);

    vector<long long> pref(n + 1, 0);
    exclusive_scan(extended.begin(), extended.end(), pref.begin(), 0LL);

    for (int i = 0; i <= n; i++) {
        if (i) {
            cout << ' ';
        }
        cout << pref[i];
    }
    cout << '\n';

    return 0;
}